tests/scripts/all.sh

```diff
@@ -986,7 +987,16 @@ component_test_psa_crypto_rsa_no_genprime() {
 
 component_test_ref_configs () {
     msg "test/build: ref-configs (ASan build)" # ~ 6 min 20s
-    CC=gcc cmake -D CMAKE_BUILD_TYPE:String=Asan .
+    # test-ref-configs works by overwriting mbedtls_config.h; this makes cmake
+    # want to re-generate generated files that depend on it, quite correctly.
```

Comment:
In the makefiles, I excluded `mbedtls_config.h` from the list of dependencies. This means that the file will incorrectly fail to be rebuilt when we add new options (or remove options), which is a rare event. But the file won't be needlessly rebuilt whenever we change the conflict, which saves a lot of hassle on a daily basis. I think we should do the same with cmake.

Reply:
I agree that makes sense for the Makefiles, as we know they don't have proper dependency resolution anyway. But the major advantage of cmake is it does (IMO at least, I mean when I use cmake it's for that reason). That was the reason I didn't do the same as you did with make. But I see your point, we're going to have a lot of pointless rebuilds. I'll think about it.

Reply:
A somewhat complicated solution would be:

1. Don't re-generate `version_features.c` if `mbedtls_config.h` isn't the checked-in version optionally modified by `config.py`. For example check if it contains `\file mbedtls_config.h`.
2. When re-generating `version_features.c`, if the resulting file is identical to the target, consider the target up-to-date: don't overwrite `version_features.c` and don't consider anything that depends on it.

I don't think it's worth it.

Reply:
> I don't think it's worth it.

Agreed. What do you think if the current state: have the dependency (which will perhaps trigger more re-generates than necessary, but at least ensures we always never miss when a rebuild is indeed needed) and keep this special case here. Is that something you could live with or do you feel strongly about removing the dependency instead?

Reply:
Personally, I pretty much don't use cmake except when reproducing an issue that was reported with it. So I don't care.

However, from the perspective of a user of Mbed TLS, I think it makes more sense not to regenerate the files, because users will very often change the configuration but not add new options. Regenerating `error.c` and `version_features.c` isn't a big deal though. I'm more concerned with running `generate_psa_tests.py` when `include/psa/crypto_config.h` has changed, because this one requires a host compiler. Consider the scenario where I want to do multiple builds for different embedded devices, each with their own configuration. I'd like to be able to generate those files with a host compiler once and for all, and then be able to do cross-compilation without having to worry about a host compiler still being available.

Reply:
Indeed that's a good point. Note however that for now we don't support `DEV_MODE` with cross-compilation, so people would have to turn if off when cross-compiling anyway. Even if we supported it, people could sill turn it off after generating the files once, but I agree that would be an inconvenience - OTOH that's a rather advanced use case so it might not be entirely unfair to expect advanced users to take few manual steps.

I don't think it's very important either way, but after your last message I'm now more inclined to just remove the dependency.